src/test/scala/chiselTests/util/PlusArgsValueSpec.scala

```diff
@@ -0,0 +1,50 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chiselTests.util
+
+import chisel3._
+import chisel3.stage.{ChiselGeneratorAnnotation, ChiselStage}
+import chisel3.testers.BasicTester
+import chisel3.util.circt.PlusArgsValue
+
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers
+
+import scala.io.Source
+
+private class PlusArgsValueTop extends Module {
+  val io = IO(new Bundle {
+    val wf = Output(UInt(1.W))
+    val wv = Output(UInt(32.W))
+    val xf = Output(UInt(1.W))
+    val xv = Output(UInt(32.W))
+  })
+  val tmpw = PlusArgsValue(UInt(32.W), "FOO=%d")
+  val tmpx = PlusArgsValue(io.xv, "BAR=%d")
```

Comment:
if we are going to support complicated bundle types then please show a test of that here... but I'm not sure the advantage of having it take any Data vs just a UInt (or SInt possibly through some API...?)

Reply:
Here, for example is parsing a non-integer substring.  We can't express this in firrtl since we don't have real types, but in verilog: `$value$plusargs("FREQ+%0F",frequency)`

you can also control integer encoding in parsing, parse strings, etc.

Reply:
can you show that being used in the test? 

how would we cast something with format string "FREQ+%0F" to a `gen: DecoupledIO(UInt(32.W))`, something the API will happily let us do right now?